docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,431 @@
+       "      <td id=\"T_3c7a2_row7_col3\" class=\"data row7 col3\" >[0, 1, 1]</td>\n",
+       "      <td id=\"T_3c7a2_row7_col4\" class=\"data row7 col4\" >1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
+       "      <td id=\"T_3c7a2_row8_col0\" class=\"data row8 col0\" >5.163767</td>\n",
+       "      <td id=\"T_3c7a2_row8_col1\" class=\"data row8 col1\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row8_col2\" class=\"data row8 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row8_col3\" class=\"data row8 col3\" >[1, 1, 1]</td>\n",
+       "      <td id=\"T_3c7a2_row8_col4\" class=\"data row8 col4\" >1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
+       "      <td id=\"T_3c7a2_row9_col0\" class=\"data row9 col0\" >4.256865</td>\n",
+       "      <td id=\"T_3c7a2_row9_col1\" class=\"data row9 col1\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row9_col2\" class=\"data row9 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row9_col3\" class=\"data row9 col3\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row9_col4\" class=\"data row9 col4\" >1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n"
+      ],
+      "text/plain": [
+       "<pandas.io.formats.style.Styler at 0x7f61c466f4f0>"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = pd.DataFrame(\n",
+    "    {\n",
+    "        \"Distance\": distance[:, 0],\n",
+    "        f\"Top{n_neighbors} (scikit-learn)\": [list(row) for row in topk_sk],\n",
+    "        \"Majority vote (scikit-learn)\": predict_sklearn,\n",
+    "        f\"Top{n_neighbors} (Concrete ML)\": [list(row) for row in topk_cml],\n",
+    "        \"Majority vote (Concrete ML)\": pred_cml_simulate,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "# Custom style function to highlight distinct rows\n",
+    "def highlight_diff(row):\n",
+    "    return [\n",
+    "        \"background-color: yellow\"\n",
+    "        if row[\"Majority vote (Concrete ML)\"] != row[\"Majority vote (scikit-learn)\"]\n",
+    "        else \"\"\n",
+    "    ] * len(row)\n",
+    "\n",
+    "df.style.apply(highlight_diff, axis=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Conclusion\n",
+    "\n",
+    "Due to its notable time and memory complexity, the K-nearest neighbors classifier is currently not well-suited for real-world applications, especially those involving large data-sets.\n",
+    "\n",
+    "Performance-wise, the K-nearest neighbors classifier matches pretty good its scikit-learn equivalent. The loss in accuracy is caused by the quantization and to the fact that the sorting algorithm is non-stable."
```

Comment:
"Performance-wise, the K-nearest neighbors classifier matches pretty good its scikit-learn equivalent."
-> 
"The K-nearest neighbors classifier matches its scikit-learn equivalent's performance."

"The loss in accuracy is caused by the quantization and to the fact that the sorting algorithm is non-stable." -> 
"The small loss in accuracy comes from quantization artifacts as well as the fact that the sorting algorithm is non-stable." (still, I would explain what you mean by non-stable here)

also, maybe insist on the fact that FHE is exact, as we usually do

Reply:
I rephrased it, btw